Tiflash crash for npe when preHandleSSTsToDTFiles

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
[2023/08/13 02:42:43.170 +08:00] [ERROR] [BaseDaemon.cpp:376] [########################################] [source=BaseDaemon] [thread_id=63]
[2023/08/13 02:42:43.170 +08:00] [ERROR] [BaseDaemon.cpp:377] ["(from thread 61) Received signal Segmentation fault(11)."] [source=BaseDaemon] [thread_id=63]
[2023/08/13 02:42:43.170 +08:00] [ERROR] [BaseDaemon.cpp:405] ["Address: NULL pointer."] [source=BaseDaemon] [thread_id=63]
[2023/08/13 02:42:43.170 +08:00] [ERROR] [BaseDaemon.cpp:413] ["Access: read."] [source=BaseDaemon] [thread_id=63]
[2023/08/13 02:42:43.170 +08:00] [ERROR] [BaseDaemon.cpp:425] ["Unknown si_code."] [source=BaseDaemon] [thread_id=63]
[2023/08/13 02:42:47.650 +08:00] [ERROR] [BaseDaemon.cpp:569] ["\n       0x75b9181\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+123441537]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:220\n  0x7f3473447df0\t<unknown symbol> [libc.so.6+347632]\n       0x80318fa\tDB::KVStore::preHandleSSTsToDTFiles(std::__1::shared_ptr<DB::Region>, DB::SSTViewVec, unsigned long, unsigned long, DB::DM::FileConvertJobType, DB::TMTContext&) [tiflash+134420730]\n                \tdbms/src/Storages/Transaction/ApplySnapshot.cpp:342\n       0x8035b3c\tDB::KVStore::handleIngestSST(unsigned long, DB::SSTViewVec, unsigned long, unsigned long, DB::TMTContext&) [tiflash+134437692]\n                \tdbms/src/Storages/Transaction/ApplySnapshot.cpp:596\n       0x80a414b\tHandleIngestSST [tiflash+134889803]\n                \tdbms/src/Storages/Transaction/ProxyFFI.cpp:453\n  0x7f3474fe3ad9\t_$LT$engine_store_ffi..observer..TiFlashObserver$LT$T$C$ER$GT$$u20$as$u20$raftstore..coprocessor..QueryObserver$GT$::post_exec_query::hb803c7bf090298f6 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+25299673]\n  0x7f347605ea3d\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::apply_raft_cmd::hb13ae1a7d2dd41c4 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+42580541]\n  0x7f3476078448\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::process_raft_cmd::h9bee616879e249ce [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+42685512]\n  0x7f347607dd94\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::handle_raft_committed_entries::h5edfa7fdf2b1d350 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+42708372]\n  0x7f34750dba32\t_$LT$raftstore..store..fsm..apply..ApplyPoller$LT$EK$GT$$u20$as$u20$batch_system..batch..PollHandler$LT$raftstore..store..fsm..apply..ApplyFsm$LT$EK$GT$$C$raftstore..store..fsm..apply..ControlFsm$GT$$GT$::handle_normal::h29727265a8d7e636 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+26315314]\n  0x7f347504a603\tbatch_system::batch::Poller$LT$N$C$C$C$Handler$GT$::poll::ha2d11ad7acb5fea2 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+25720323]\n  0x7f347514b7a0\tstd::sys_common::backtrace::__rust_begin_short_backtrace::h7d8b9e58297764f6 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+26773408]\n  0x7f347518bfee\tcore::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h29e7d480a13dc6b4 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+27037678]\n  0x7f3476893835\tstd::sys::unix::thread::Thread::new::thread_start::hd2791a9cabec1fda [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+51185717]\n                \t/rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/std/src/sys/unix/[thread.rs:108](http://thread.rs:108/)\n  0x7f3473492802\tstart_thread [libc.so.6+653314]"] [source=BaseDaemon] [thread_id=63]
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
master

--- a/dbms/src/Storages/Transaction/KVStore.h
+++ b/dbms/src/Storages/Transaction/KVStore.h
@@ -261,14 +261,14 @@ private:
         std::shared_ptr<std::atomic_bool> registerTask(uint64_t region_id)
         {
             // Automaticlly override the old one.
-            genLockGuard();
+            auto _ = genLockGuard();
             auto b = std::make_shared<std::atomic_bool>(false);
             tasks[region_id] = b;
             return b;
         }
         std::shared_ptr<std::atomic_bool> deregisterTask(uint64_t region_id)
         {
-            genLockGuard();
+            auto _ = genLockGuard();
             auto it = tasks.find(region_id);
             if (it != tasks.end())
             {

--- a/dbms/src/Storages/Transaction/KVStore.h
+++ b/dbms/src/Storages/Transaction/KVStore.h
@@ -94,7 +94,11 @@ public:
 
     void gcPersistedRegion(Seconds gc_persist_period = Seconds(60 * 5));
 
-    static bool tryFlushRegionCacheInStorage(TMTContext & tmt, const Region & region, const LoggerPtr & log, bool try_until_succeed = true);
+    static bool tryFlushRegionCacheInStorage(
+        TMTContext & tmt,
+        const Region & region,
+        const LoggerPtr & log,
+        bool try_until_succeed = true);
 
     size_t regionSize() const;
     EngineStoreApplyRes handleAdminRaftCmd(
@@ -112,12 +116,25 @@ public:
         TMTContext & tmt) const;
 
     bool needFlushRegionData(UInt64 region_id, TMTContext & tmt);
-    bool tryFlushRegionData(UInt64 region_id, bool force_persist, bool try_until_succeed, TMTContext & tmt, UInt64 index, UInt64 term);
+    bool tryFlushRegionData(
+        UInt64 region_id,
+        bool force_persist,
+        bool try_until_succeed,
+        TMTContext & tmt,
+        UInt64 index,
+        UInt64 term);
 
     /**
      * Only used in tests. In production we will call preHandleSnapshotToFiles + applyPreHandledSnapshot.
      */
-    void handleApplySnapshot(metapb::Region && region, uint64_t peer_id, SSTViewVec, uint64_t index, uint64_t term, std::optional<uint64_t>, TMTContext & tmt);
+    void handleApplySnapshot(
+        metapb::Region && region,
+        uint64_t peer_id,
+        SSTViewVec,
+        uint64_t index,
+        uint64_t term,
+        std::optional<uint64_t>,
+        TMTContext & tmt);
 
     void handleIngestCheckpoint(RegionPtr region, CheckpointInfoPtr checkpoint_info, TMTContext & tmt);
 
@@ -187,7 +204,14 @@ private:
     friend struct MockRaftCommand;
     friend class RegionMockTest;
     friend class NaturalDag;
-    friend void RegionBench::concurrentBatchInsert(const TiDB::TableInfo &, Int64, Int64, Int64, UInt64, UInt64, Context &);
+    friend void RegionBench::concurrentBatchInsert(
+        const TiDB::TableInfo &,
+        Int64,
+        Int64,
+        Int64,
+        UInt64,
+        UInt64,
+        Context &);
     using DBGInvokerPrinter = std::function<void(const std::string &)>;
     friend void dbgFuncRemoveRegion(Context &, const ASTs &, DBGInvokerPrinter);
     friend void dbgFuncPutRegion(Context &, const ASTs &, DBGInvokerPrinter);
@@ -220,16 +244,22 @@ private:
     template <typename RegionPtrWrap>
     void onSnapshot(const RegionPtrWrap &, RegionPtr old_region, UInt64 old_region_index, TMTContext & tmt);
 
-    RegionPtr handleIngestSSTByDTFile(const RegionPtr & region, const SSTViewVec, UInt64 index, UInt64 term, TMTContext & tmt);
+    RegionPtr handleIngestSSTByDTFile(
+        const RegionPtr & region,
+        const SSTViewVec,
+        UInt64 index,
+        UInt64 term,
+        TMTContext & tmt);
 
     // Remove region from this TiFlash node.
     // If region is destroy or moved to another node(change peer),
     // set `remove_data` true to remove obsolete data from storage.
-    void removeRegion(RegionID region_id,
-                      bool remove_data,
-                      RegionTable & region_table,
-                      const KVStoreTaskLock & task_lock,
-                      const RegionTaskLock & region_lock);
+    void removeRegion(
+        RegionID region_id,
+        bool remove_data,
+        RegionTable & region_table,
+        const KVStoreTaskLock & task_lock,
+        const RegionTaskLock & region_lock);
     void mockRemoveRegion(RegionID region_id, RegionTable & region_table);
     KVStoreTaskLock genTaskLock() const;
 
@@ -247,10 +277,26 @@ private:
     /// Notice that if flush_if_possible is set to false, we only check if a flush is allowed by rowsize/size/interval.
     /// It will not check if a flush will eventually succeed.
     /// In other words, `canFlushRegionDataImpl(flush_if_possible=true)` can return false.
-    bool canFlushRegionDataImpl(const RegionPtr & curr_region_ptr, UInt8 flush_if_possible, bool try_until_succeed, TMTContext & tmt, const RegionTaskLock & region_task_lock, UInt64 index, UInt64 term);
-    bool forceFlushRegionDataImpl(Region & curr_region, bool try_until_succeed, TMTContext & tmt, const RegionTaskLock & region_task_lock, UInt64 index, UInt64 term) const;
+    bool canFlushRegionDataImpl(
+        const RegionPtr & curr_region_ptr,
+        UInt8 flush_if_possible,
+        bool try_until_succeed,
+        TMTContext & tmt,
+        const RegionTaskLock & region_task_lock,
+        UInt64 index,
+        UInt64 term);
+    bool forceFlushRegionDataImpl(
+        Region & curr_region,
+        bool try_until_succeed,
+        TMTContext & tmt,
+        const RegionTaskLock & region_task_lock,
+        UInt64 index,
+        UInt64 term) const;
 
-    void persistRegion(const Region & region, std::optional<const RegionTaskLock *> region_task_lock, const char * caller) const;
+    void persistRegion(
+        const Region & region,
+        std::optional<const RegionTaskLock *> region_task_lock,
+        const char * caller) const;
     void releaseReadIndexWorkers();
     void handleDestroy(UInt64 region_id, TMTContext & tmt, const KVStoreTaskLock &);
 

